pkg/controller/remote/cluster.go

```diff
@@ -25,23 +25,30 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// ClusterClientInterface is an interface encapsulating methods
+// to access a remote cluster.
+type ClusterClientInterface interface {
+	RESTConfig() *restclient.Config
```

Comment:
Do we need to include this?

Reply:
I added it for testing later

Reply:
I typically recommend avoiding adding things to the interface that aren't needed by real consumers. If you need to get at the rest config in a unit test, I'd recommend a type assertion (e.g. `c.(*ClusterClient).restConfig`).

Reply:
The interface would help in case I need to mock the ClusterClient object with a fake one instead. I still need to get to testing phase though, if I don't need this interface, I'm happy to remove it